runner.go

```diff
@@ -219,6 +256,11 @@ func (r *Runner) Start() {
 			log.Printf("[INFO] (runner) received template %q from quiescence", tmpl.Path)
 			delete(r.quiescenceMap, tmpl.Path)
 
+		case c := <-childExitCh:
+			log.Printf("[INFO] (runner) child process died")
+			r.ErrCh <- fmt.Errorf("child process died with exit code %d", c)
```

Comment:
I don't think so - the assumption is that the child process continues to run in the foreground. If the process stops (even with exit 0 and success), it's no longer under management and therefore Consul Template cannot send signals and manage that processes' lifecycle.